catalog/app/components/Preview/loaders/IFrame/LoaderBrowsable.tsx

```diff
@@ -123,71 +132,58 @@ function useSession(handle: FileHandle) {
 
   React.useEffect(() => {
     let ignore = false
-    let sessionClosure: Session | null = null
+    let sessionId: SessionId = ''
+    let timer: NodeJS.Timer
 
-    async function requestSession() {
+    async function initSession() {
       try {
-        const s = await createSession(scope, SESSION_TTL)
+        const session = await createSession(scope, SESSION_TTL)
         if (ignore) return
-        sessionClosure = s
-        setSession(s)
+        sessionId = session.id
+        setResult(AsyncResult.Ok(sessionId))
+
+        timer = setInterval(async () => {
+          try {
+            if (!sessionId) return
+            await refreshSession(sessionId, SESSION_TTL)
+          } catch (e) {
+            clearInterval(timer)
+            setResult(AsyncResult.Err(mapPreviewError(retry, e as ErrorLike)))
```

Comment:
there should be a check for `ingore` here as well, bc the response can come after the cleanup callback invocation